### PR TITLE
Valid Moves for Knight: ready to merge!

### PR DESCRIPTION
**Pull request by ryanveroniwooff**
Will someone look this over and make sure I did that correctly. thanks!

### CODE REVIEW

File: app/models/knight.rb [hunk 1]
```diff
@@ -1,2 +1,13 @@
...
+  def valid_move?(destination_x, destination_y)
+    return false if x_position == destination_x
+    return false if y_position == destination_y
+    valid_length?(destination_x, destination_y)
+  end
+
+  private
+
+  def valid_length?(destination_x, destination_y)
+    (x_position - destination_x).abs + (y_position - destination_y).abs == 3
```
Comment (mscoutermarsh) [inline app/models/knight.rb]: Would this allow someone to move backwards?

File: app/models/knight.rb [hunk 2]
```diff
@@ -1,2 +1,13 @@
 class Knight < Piece
+  def valid_move?(destination_x, destination_y)
+    return false if x_position == destination_x
+    return false if y_position == destination_y
+    valid_length?(destination_x, destination_y)
+  end
```
Comment (mscoutermarsh) [inline app/models/knight.rb]: very nice method

File: app/models/knight.rb [hunk 3]
```diff
@@ -1,2 +1,13 @@
...
+
+  private
+
+  def valid_length?(destination_x, destination_y)
```
Comment (mscoutermarsh) [inline app/models/knight.rb]: I'd rename this to something more descriptive, maybe `valid_move_distance?`

File: spec/models/knight_spec.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    let(:game) { FactoryGirl.create(:game) }
+    let(:knight) { FactoryGirl.create(:knight, x_position: 1, y_position: 0, game: game) }
+
+    context 'when not valid destination' do
+      it { expect(knight.valid_move?(3, 4)).to eq false }
+    end
+
+    context 'when valid' do
+      it { expect(knight.valid_move?(3, 1)).to eq true }
+    end
```
Comment (mscoutermarsh) [inline spec/models/knight_spec.rb]: :+1:

File: app/models/knight.rb (hunk 1 above)
Comment (ryanveroniwooff) [inline app/models/knight.rb]: Yes, I believe so. I thought the Knight could move in any direction?

Comment (mscoutermarsh) [inline app/models/knight.rb]: :+1:

**Pull request merged**